Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -13,10 +13,15 @@ private enum SearchListSection: Int {
     case Search
 }
 
+protocol SearchViewControllerDelegate: class {
```

Comment:
> All of these table views we have are ways to show lists of urls or sites. I guess I don't see them as attached to views, but if you wanted to view it that way, they're all attached to SiteTableViewControllers. We did that intentionally because they all support shared display types and shared functionality.

But we can't just replace `HomePanel` with a `SiteTableViewControllerDelegate`. That unnecessarily restricts panels to be a certain kind of view. Specifically, this abstraction will break once we introduce a top sites panel since that won't be a `SiteTableViewController`.

> These ViewControllers shouldn't care where they're displayed, if its on Home or in some other place. They're not really HomePanels. HomePanels are (IMO) just a convenience for generating these ViewControllers and maintaining a list of which ones to show on about:home.

So what are they? `SiteTableViewControllerDelegate` is too specific for the reasons mentioned above. We could create another layer of abstraction, a `SiteViewController` and a `SiteViewControllerDelegate`, then have `SiteTableViewController` subclass `SiteViewController`, then have the panels and `SearchViewController` all use `SiteViewControllerDelegate`. Would that work for you? I'm just not sure what that buys us -- we could always create this abstraction layer later if we actually need it (which I bet we won't).

To see the problem I'm trying to solve now, take a look at [this line](https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Browser/BrowserViewController.swift#L377) in BVC. Is that for handling the history or bookmarks panel? Nope, those are handled by the `HomePanelViewControllerDelegate`. The only thing it does handle is the `SearchViewController`. And that's not even a home panel! The abstraction is clearly broken is as.